### PR TITLE
Port android.widget.CompoundButton to RxJava 2.

### PR DESCRIPTION
**Pull request by ghostbuster91**

### CODE REVIEW

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/CompoundButtonCheckedChangeObservable.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    protected void subscribeActual(Observer<? super Boolean> observer) {
+        verifyMainThread();
+        Listener listener = new Listener(view, observer);
+        observer.onSubscribe(listener);
+        view.setOnCheckedChangeListener(listener);
+    }
+
+    private static final class Listener extends MainThreadDisposable implements CompoundButton.OnCheckedChangeListener {
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/CompoundButtonCheckedChangeObservable.java]: Drop `private`

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/CompoundButtonCheckedChangeObservable.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        private final CompoundButton view;
+        private final Observer<? super Boolean> observer;
+
+        private Listener(CompoundButton view, Observer<? super Boolean> observer) {
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/CompoundButtonCheckedChangeObservable.java]: Drop `private`

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/CompoundButtonCheckedChangeObservable.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            if (!isDisposed()) {
+                observer.onNext(isChecked);
+            }
+        }
+
+        @Override
+        protected void onDispose() {
+            view.setOnCheckedChangeListener(null);
+        }
+    }
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/CompoundButtonCheckedChangeObservable.java]: This whole file should be two-space indent.

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxCompoundButton.java [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
+
+
+import io.reactivex.Observable;
+import io.reactivex.functions.Consumer;
 
 import static com.jakewharton.rxbinding.internal.Preconditions.checkNotNull;
 
 /**
- * Static factory methods for creating {@linkplain Observable observables} and {@linkplain Action1
+ * Static factory methods for creating {@linkplain Observable observables} and {@linkplain io.reactivex.functions.Consumer
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxCompoundButton.java]: nit: this imported so you don't need to qualify it

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxCompoundButton.java [hunk 5]
```diff
@@ -27,21 +29,23 @@
   @CheckResult @NonNull
   public static Observable<Boolean> checkedChanges(@NonNull CompoundButton view) {
     checkNotNull(view, "view == null");
-    return Observable.create(new CompoundButtonCheckedChangeOnSubscribe(view));
+    return new CompoundButtonCheckedChangeObservable(view);
   }
 
+
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxCompoundButton.java]: delete

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxCompoundButton.java [hunk 6]
```diff
@@ -27,21 +29,23 @@
...
-  public static Action1<? super Boolean> checked(@NonNull final CompoundButton view) {
+  public static Consumer<? super Boolean> checked(@NonNull final CompoundButton view) {
     checkNotNull(view, "view == null");
-    return new Action1<Boolean>() {
-      @Override public void call(Boolean value) {
-        view.setChecked(value);
+    return new Consumer<Boolean>(){
+      @Override
+      public void accept(Boolean value) throws Exception {
+          view.setChecked(value);
```
Comment (JakeWharton) [inline rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxCompoundButton.java]: nit: two-space indent

**Comment by ghostbuster91** [comment]
@JakeWharton Done :) Btw could you explain why this `private` modifier had to be removed from listener?

**Comment by JakeWharton** [comment]
It avoids the need for a synthetic accessor method to be generated.

**Pull request merged**